metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanAspectDao.java

```diff
@@ -597,6 +628,41 @@ public <T> T runInTransactionWithRetry(
       @Nonnull final Function<Transaction, T> block,
       @Nullable AspectsBatch batch,
       final int maxTransactionRetry) {
+
+    if (locks != null && batch != null) {
+      Set<Urn> urnsWithKeyAspects =
+          batch.getMCPItems().stream()
+              .filter(i -> i.getEntitySpec().getKeyAspectSpec().equals(i.getAspectSpec()))
+              .map(MCPBatchItem::getUrn)
+              .collect(Collectors.toSet());
+
+      if (!urnsWithKeyAspects.isEmpty()) {
+        for (AspectsBatch splitBatch : splitByUrn(batch, urnsWithKeyAspects)) {
+          try {
+            Lock lock =
+                locks.get(splitBatch.getMCPItems().stream().findFirst().get().getUrn().toString());
```

Comment:
So this will create locks for parts of the batch that don't have key aspects too? Don't think it's really a concern, just checking.

Reply:
Yeah, that `other` batch should bypass this lock, but in reality today we almost never batch more than 1 urn so this is a concern when we do throw multiple at it. The one place where we do this is actually the snapshot ingestion path. I'll address it here though anyway. 

@RyanHolstien - Nice catch!!